libs/state/actions/docs/Readme.md

```diff
@@ -0,0 +1,307 @@
+
+Subjects are normally used to implement this feature. This leads, especialls in bigger applications to messy code that is bloated with Subjects.
+
+Let's have a look at this piece of code:
+
+```typescript
+@Component({
+template: `
+<input (input)="searchInput($event?.target?.value)" /> Search for: {{search$ | async}}<br/>
+<button (click)="submitBtn()">Submit<button><br/>
+<ul>
+  <li *ngFor="let item of list$ | async as list">{{item}}</li>
+</ul>
+`})
+class Component {
+  private _submitBtn = new Sunject<void>();
+  private _searchInput = new Sunject<string>();
+    
+  set submitBtn() {
+    _submitBtn.next()
+  }
+  get submitBtn$() {
+    return _searchInput.asObservable();
+  }
+  set search(search: string) {
+    _searchInput.next(search)
+  }
+  get search$() {
+    return _searchInput.asObservable();
+  }
+    
+  list$ = this.submitBtn$.pipe(
+    withLatestFrom(this.search$),
+    switchMap(([_, searchString]) => api.query(searchString))
+   )
+  
+  consotructor(api: API) {
+  
+  }
+
+}
+```
+
+Just look at the amount of code we need to write for only 2 observable ui events.
+
+Downsides:
+- Boiler plate for setter and getter
+- Transfromations spreaded over class and template
+- Manual typing ov subjects and streams
+- no central plae for ui trigger
+
+Imagine we could have a configurable functions that retuens all ui logic typed under one object.
+
+```typescript
+import { RxActionsFactory } from './actions.factory'; 
+interface UiActions {
+ submitBtn: void,
+ searchInput: string
+}
+
+@Component({
+template: `
+<input (input)="ui.searchInput($event)" /> Search for: {{ui.search$ | async}}<br/>
+<button (click)="ui.submitBtn()">Submit<button><br/>
+<ul>
+  <li *ngFor="let item of list$ | async as list">{{item}}</li>
+</ul>
+`,
+providers: [RxActionsFactory]
+})
+class Component {
+  ui = factory.create({searchInput: (e) => e?.target?.value});
+  
+  list$ = this.ui.submitBtn$.pipe(
+    withLatestFrom(this.ui.search$),
+    switchMap(([_, searchString]) => api.query(searchString))
+   )
+  
+  consotructor(api: API, 
+   private factory: RxActionsFactory<UiActions>
+) {
+  
+  }
+
+}
+```
+
+# RxAngular Actions
+
+This package helps to reduce code used to create composable action streams. 
+It mostly is used in combination with state management libs to handle user interaction and backend communication.
+
+## Setup
+
+The coalescing features can be used directly from the `cdk` package or indirectly through the `state` package.
+To do so, install the `cdk` package and, if needed, the packages depending on it:
+
+1. Install `@rx-angular/state`
+
+```bash
+npm i @rx-angular/state
+// or
+yarn add @rx-angular/state
+```
+
+## Usage
+
+As actions are used in global state management as well as in UI event composition we will have a look on both ends.
+
+
+There are 2 common ways to create an action stream and its emission:
```

Comment:
I find this section a bit confusing, we expect the usage of the library and not alternatives without.